corehq/apps/change_feed/pillow.py

```diff
@@ -44,20 +44,30 @@ def process_change(self, pillow_instance, change, do_set_checkpoint=False):
                 domain=change.document.get('domain', None),
                 is_deletion=change.deleted,
             )
-            try:
+
+            def _send_to_kafka():
+                # closures
                 self._producer.send_messages(
                     bytes(get_topic(document_type)),
                     bytes(change_meta.domain.encode('utf-8') if change_meta.domain is not None else None),
                     bytes(json.dumps(change_meta.to_json())),
                 )
+
+            try:
+                try:
+                    _send_to_kafka()
+                except FailedPayloadsError:
+                    # this typically an inactivity timeout - which can happen if the feed is low volume
+                    # just do a simple retry
+                    _send_to_kafka()
             except LeaderNotAvailableError:
                 # kafka seems to be down. sleep a bit to avoid crazy amounts of error spam
                 time.sleep(15)
```

Comment:
heh, it delays the rate of errors from arbitrarily many per second to one per 15 seconds max. if this fires something is likely very wrong so i didn't put too much thought into it